iothub/service/src/IoTHubSasCredential.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+#if !NET451
+
+using Azure;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Shared access signature credential used to authenticate with IoT hub.
+    /// </summary>
+    public class IoTHubSasCredential : AzureSasCredential
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="IoTHubSasCredential"/>.
+        /// </summary>
+        /// <param name="signature">Shared access signature used to authenticate with IoT hub.</param>
+        /// <param name="expiresOnUtc">The shared access signature expiry in UTC.</param>
+        public IoTHubSasCredential(string signature, DateTime expiresOnUtc) : base(signature)
+        {
+            ExpiresOnUtc = expiresOnUtc;
+        }
+
+        /// <summary>
+        /// Updates the shared access signature. This is intended to be used when you've
+        /// regenerated your shared access signature and want to update clients.
+        /// </summary>
+        /// <param name="signature">Shared access signature used to authenticate with IoT hub.</param>
+        /// <param name="expiresOnUtc">The shared access signature expiry in UTC.</param>
+        public void Update(string signature, DateTime expiresOnUtc)
+        {
+            Update(signature);
+            ExpiresOnUtc = expiresOnUtc;
+        }
+
+        /// <summary>
+        /// The shared access signature expiry in UTC.
+        /// </summary>
+        public DateTime ExpiresOnUtc { get; private set; }
```

Comment:
The one qualm I have with this is that the signature already has this this information baked in as one of the key value pairs, so it feels a bit redundant. I get that parsing it out and converting seconds to a DateTime field leave us open to bugs, but I'm curious to hear everyone's opinion on this.

Reply:
Also, should this field be a DateTimeOffset, since it is supposed to be UTC?

Reply:
The AMQP library expects a DateTime in UTC so I kept it as DateTime. We don't need to do unnecessary conversions.